src/hooks/useFingerprint.ts

```diff
@@ -1,14 +1,36 @@
 'use client';
+import FingerprintJS from '@fingerprintjs/fingerprintjs';
 import { useEffect, useState } from 'react';
 
 export const useFingerprint = () => {
-  const [fingerprint, setFingerprint] = useState<string>();
+  const [fingerprint, setFingerprint] = useState<string | undefined>(() => {
+    if (
+      typeof window !== 'undefined' &&
+      typeof sessionStorage !== 'undefined'
+    ) {
+      return sessionStorage.getItem('fpId') || undefined;
+    }
+    return undefined;
+  });
 
   useEffect(() => {
-    // Get the client's fingerprint id;
-    // const newFingerprint = 'abcd';
-    // console.log(newFingerprint);
-    // setFingerprint(`${newFingerprint}`);
+    let active = true;
+    load();
+    return () => {
+      active = false;
```

Comment:
This is a race condition as you are setting a variable which can be destroyed before it, read https://legacy.reactjs.org/docs/hooks-effect.html#example-using-hooks-1 usEffect return